pkg/ingester/user_metrics_metadata_test.go

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ingester
+
+import (
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+func TestUserMetricsMetadata(t *testing.T) {
```

Comment:
I've run the test locally and it looks flaky. For example, if I run it 10 times, sometimes it fails:

```
go test -run 'TestUserMetricsMetadata' -count=10 ./pkg/ingester

--- FAIL: TestUserMetricsMetadata (0.00s)
    --- FAIL: TestUserMetricsMetadata/should_fail_when_metadata_per_user_limit_reached (0.00s)
        user_metrics_metadata_test.go:97:
            	Error Trace:	user_metrics_metadata_test.go:97
            	Error:      	An error is expected but got nil.
            	Test:       	TestUserMetricsMetadata/should_fail_when_metadata_per_user_limit_reached
--- FAIL: TestUserMetricsMetadata (0.00s)
    --- FAIL: TestUserMetricsMetadata/should_fail_when_metadata_per_user_limit_reached (0.00s)
        user_metrics_metadata_test.go:97:
            	Error Trace:	user_metrics_metadata_test.go:97
            	Error:      	An error is expected but got nil.
            	Test:       	TestUserMetricsMetadata/should_fail_when_metadata_per_user_limit_reached
--- FAIL: TestUserMetricsMetadata (0.00s)
    --- FAIL: TestUserMetricsMetadata/should_fail_when_metadata_per_user_limit_reached (0.00s)
        user_metrics_metadata_test.go:97:
            	Error Trace:	user_metrics_metadata_test.go:97
            	Error:      	An error is expected but got nil.
            	Test:       	TestUserMetricsMetadata/should_fail_when_metadata_per_user_limit_reached
--- FAIL: TestUserMetricsMetadata (0.00s)
    --- FAIL: TestUserMetricsMetadata/should_fail_when_metadata_per_metric_limit_reached (0.00s)
        user_metrics_metadata_test.go:97:
            	Error Trace:	user_metrics_metadata_test.go:97
            	Error:      	An error is expected but got nil.
            	Test:       	TestUserMetricsMetadata/should_fail_when_metadata_per_metric_limit_reached
--- FAIL: TestUserMetricsMetadata (0.00s)
    --- FAIL: TestUserMetricsMetadata/should_fail_when_metadata_per_metric_limit_reached (0.00s)
        user_metrics_metadata_test.go:97:
            	Error Trace:	user_metrics_metadata_test.go:97
            	Error:      	An error is expected but got nil.
            	Test:       	TestUserMetricsMetadata/should_fail_when_metadata_per_metric_limit_reached
--- FAIL: TestUserMetricsMetadata (0.00s)
    --- FAIL: TestUserMetricsMetadata/should_fail_when_metadata_per_user_limit_reached (0.00s)
        user_metrics_metadata_test.go:97:
            	Error Trace:	user_metrics_metadata_test.go:97
            	Error:      	An error is expected but got nil.
            	Test:       	TestUserMetricsMetadata/should_fail_when_metadata_per_user_limit_reached
FAIL
FAIL	github.com/grafana/mimir/pkg/ingester	0.392s
FAIL
```

Reply:
I think the reason is that you're using a `map` for `metadataToExpectError`. When iterating a map, there's no ordering guaranteed in golang. You should change it to a slice.

Reply:
🤦 thanks for catching this